### PR TITLE
Add gitbook-plugin-offline

### PR DESCRIPTION
**Pull request by lwakefield**
This PR adds the `gitbook-plugin-offline` which uses service workers to make the content available offline.

### CODE REVIEW

**Comment by RX14** [comment]
If we're using `gitbook install` we shouldn't add it to the `package.json`, same for the `ga` plugin. Shall we remove the `package.json` entirely?

**Comment by straight-shoota** [comment]
Using `gitbook install` is recommended for this plugin. So you don't need to add it as a dev dependency in `package.json`.

I'm not sure if we should or could remove `package.json` entirely though. because it also contains project metadata.

File: package.json [hunk 1]
```diff
@@ -4,6 +4,7 @@
   "license": "CC0",
   "repository": "github:crystal-lang/crystal-book",
   "devDependencies": {
-    "gitbook-plugin-ga": "^1.0.1"
+    "gitbook-plugin-ga": "^1.0.1",
+    "gitbook-plugin-offline": "0.0.3"
```
Comment (straight-shoota) [inline package.json]: No need to add the plugin here, `gitbook install` will take care of this.

Comment (straight-shoota) [inline package.json]: No need to add the plugin here, `gitbook install` will take care of this.

Comment (lwakefield) [reply package.json]: Addressed in cb159c8

**Pull request merged**